enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,374 @@
+
+In summary although the above workflow is feasible, the present proposal does **not** recommend any further implementation leaving users to manage their own `ClusterLogForwarder` resource per namespace if needed, because:
+1. The translation of per namespace `LoggingStack` resources to `ClusterLogForwarder` is of diminishing returns.
+2. More important the limitations of Server-Side-Apply on list items make several annotation makers on the `ClusterLogForwarder` resource mandatory (See later on [SSA-Limitations](#limitations-of-server-side-apply-on-openshift-logging-resources))
+
+#### Implications of Server-Side-Apply on OpenShift Logging resources
+
+The Cluster Observability Operator's design is centered around the Kubernetes Server-Side-Apply capability. In short it allows sharing resources between separate operator on the resource field level. However each OpenShift Logging resource has mandatory requirements on field when created. These fields will inevitably change management ownership from CLO/LO to the Cluster Obserbability Operator.
+
+In detail the following `ClusterLogging` fields will be managed by COO exlusively without allowing user edits:
+```yaml
+spec:
+  managementState: Managed
+  logStore:
+    type: lokistack
+    lokistack:
+      name: <name-of-lokistack-resource>a
+  collection:
+    type: vector
+```
+
+Also the following `LokiStack` fields will be managed by COO and allowing user edits through the `LoggingStack` resource:
+```yaml
+spec:
+  storage:
+    size: 1x.demo
+    storageClassName: gp3-csi
+    storage:
+      secret:
+        name: test
+        type: s3
+```
+
+#### Limitations of Server-Side-Apply on OpenShift Logging resources
+
+When using Server-Side-Apply the implementation needs to be informed of several caveats:
+1. Fields that have default values will inevitably show up being managed by the Cluster Observability Operator.
+2. List fields missing CRD markers such as OpenAPI `x-kubernetes-list-type` and `x-kubernetes-list-map-keys` the entire list is replaced which means the full list is managed by the Cluster Observability Operator.
+
+The first caveat is not relevant because neither `ClusterLogging` nor `LokiStack` resource make use of the `+kubebuilder:default` CRD marker in fields required for creation. The second once has severe impact on when managing `ClusterLogForwarder` resource. As per now the `ClusterLogForwarder` spec consists mainly of list fields namely `inputs`, `outputs` and `pipelines` where none is annotated with the OpenAPI `x-kubernetes-list-type` and `x-kubernetes-list-map-keys` markers. Therefore managing a `ClusterLogForwarder` via `LoggingStack` effectively means re-exposing the whole API which is considered harmful and on the opposite of the entire proposal.
+
+However, if future enhancements address use cases relevant for `LoggingStack` resources owning `ClusterLogForwarder` resources, then it will need the following markers:
```

Comment:
I'm not sure to understand: does it mean that it should be feasible to modify the CRD to have multiple actors manipulating distinct items in the lists?

Reply:
Yes this is feasible with the proper annotations as listed below. As long as you can select in the list/map item a field that can be used as identity (e.g. `name`) you can treat the list items like other crd fields in terms of ownership.

Reply:
Sorry I meant: is it something that you plan to change in the `ClusterLogForwarder` CRD?